### PR TITLE
token unlock with unbond period

### PR DESCRIPTION
**Pull request by psorinionut**
- Tokens now have an unbonding period upon unlocking
- Added an extra test for the farm boosted rewards computation

### CODE REVIEW

File: locked-asset/energy-factory/src/lib.rs [hunk 1]
```diff
@@ -161,13 +165,23 @@ pub trait SimpleLockEnergy:
...
-            &output_payment.token_identifier,
+
+        let token_payment = EsdtTokenPayment::new(
+            output_payment.token_identifier.clone(),
             0,
-            &output_payment.amount,
+            output_payment.amount.clone(),
         );
 
+        let unbond_epochs = self.unbond_epochs().get();
```
Comment (sasurobert) [inline locked-asset/energy-factory/src/lib.rs:175]: we do not need to lock for unbonding if FEEs are 0.

File: common/modules/token_unstake/src/token_unstake.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+pub const DEFAULT_UNBOND_EPOCHS: u64 = 10;
+
+#[derive(ManagedVecItem, TopEncode, TopDecode, NestedEncode, NestedDecode, TypeAbi, Clone)]
+pub struct UnstakePair<M: ManagedTypeApi> {
+    pub unlock_epoch: u64,
+    pub token_payment: EsdtTokenPayment<M>,
+}
+
+#[elrond_wasm::module]
+pub trait TokenUnstakeModule: token_send::TokenSendModule {
```
Comment (sasurobert) [inline common/modules/token_unstake/src/token_unstake.rs:15]: move to a new contract.

File: locked-asset/energy-factory/src/lib.rs (hunk 1 above)
Comment (psorinionut) [reply locked-asset/energy-factory/src/lib.rs]: Updated.

File: common/modules/token_unstake/src/token_unstake.rs (hunk 2 above)
Comment (psorinionut) [reply common/modules/token_unstake/src/token_unstake.rs:15]: Will be done in another PR.

**Pull request closed without merging**